metrics.go

```diff
@@ -144,6 +144,7 @@ func (m *metrics) registerInstance() {
 		m.err(err)
 		return
 	}
+	resp.Body.Close()
```

Comment:
Ah good catch. I forgot about this since we don't actually read the response body. I think we should maybe put a `defer` in front of this call though? Otherwise if the previous line enters the error case and returns, we still have the same problem.

Reply:
Sure. There is probably no harm in deferring this. It could prevent unintended problems if we were ever to read from the body.